sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -2938,20 +2940,20 @@ case class Extract(field: Expression, source: Expression, replacement: Expressio
 object Extract {
   def createExpr(funcName: String, field: Expression, source: Expression): Expression = {
     // both string and null literals are allowed.
-    if ((field.dataType == StringType || field.dataType == NullType) && field.foldable) {
-      val fieldStr = field.eval().asInstanceOf[UTF8String]
```

Comment:
You do not need to change this here to pattern match, you can use `field.dataType.isInstanceOf[StringType]`